### PR TITLE
Publish changes for HttpPlatformHandler

### PR DESCRIPTION
**Pull request by davidfowl**
- Added `--iis-command` switch to publish that determines what `.cmd` file (associated with a command) goes into the `processPath` section of `web.config`
- Support config merging (if existing `web.config` exists)
- Changed `.cmd` scripts to find a DNX of the same version in `%DNX_HOME%` (not done for the .sh scripts yet because it's not critical path)
- Added tests

#2682

### CODE REVIEW

File: misc/HelloWorld/src/HelloWorld/Program.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Reflection;
+using Newtonsoft.Json;
+
+namespace HelloWorld
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World");
```
Comment (anurse) [inline misc/HelloWorld/src/HelloWorld/Program.cs]: It's `Hello, World`. You're missing a comma.

:trollface:

File: src/Microsoft.Dnx.Tooling/Publish/PublishRoot.cs [hunk 2]
```diff
@@ -97,38 +99,51 @@ private void GenerateBatchFiles()
...
+                var cmdScript = $@"
+@echo off
+SET DNX_FOLDER={runtimeFolder}
+SET ""LOCAL_DNX=%~dp0{AppRootName}\runtimes\%DNX_FOLDER%\bin\{Runtime.Constants.BootstrapperExeName}.exe""
+
+IF EXIST %LOCAL_DNX% (
+  SET ""DNX_PATH=%LOCAL_DNX%""
+)
+
+for %%a in (%DNX_HOME%) do (
```
Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Publish/PublishRoot.cs]: File a bug to write automated tests that test the behavior of this script.

File: src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs [hunk 3]
```diff
@@ -516,174 +512,102 @@ private void GenerateWebConfigFileForWwwRootOut(PublishRoot root, Runtime.Projec
...
-        {
-            var child = parent.Elements().Where(x => x.Name == name).FirstOrDefault();
-            if (child == null)
-            {
-                child = new XElement(name);
-                parent.Add(child);
-            }
-            return child;
-        }
+            root.Reports.Information.WriteLine($"Using command '{command}' as the entry point for web.config.");
```
Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs]: Using command '{command}' as the entry point for<del> web.config</del> IIS.

File: src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs [hunk 4]
```diff
@@ -516,174 +512,102 @@ private void GenerateWebConfigFileForWwwRootOut(PublishRoot root, Runtime.Projec
...
+  </httpPlatform>
+</system.webServer></configuration>");
 
-        // Expected runtime name format: dnx-{FLAVOR}-{OS}-{ARCHITECTURE}.{VERSION}
-        // Sample input: dnx-coreclr-win-x86.1.0.0.0
-        // Sample output: coreclr
-        private static string GetRuntimeFlavor(PublishRuntime runtime)
-        {
-            if (runtime == null)
+            var attributesToOverwrite = new[]
```
Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs]: <del>`attributesToOverwrite`</del> `requiredAttributes` maybe? Overwrite makes me think they are the attributes that the user *can* overwrite.

File: src/Microsoft.Dnx.Tooling/Publish/PublishRoot.cs [hunk 5]
```diff
@@ -155,25 +179,9 @@ private void GenerateBashScripts()
 done
 DIR=""$( cd -P ""$( dirname ""$SOURCE"" )"" && pwd )""
 
-exec ""{1}{2}"" --appbase ""$DIR/{0}"" Microsoft.Dnx.ApplicationHost --configuration {3} {4} ""$@""";
+exec ""{runtimeFolder}{Runtime.Constants.BootstrapperExeName}"" --project ""$DIR/{relativeAppBase}"" --configuration {Configuration} {commandName} ""$@""";
```
Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Publish/PublishRoot.cs]: Bash doesn't get the "try DNX_HOME first" logic?

Comment (anurse) [inline src/Microsoft.Dnx.Tooling/Publish/PublishRoot.cs]: It should be much easier ;P

File: test/Microsoft.Dnx.CommonTestUtils/DirTree.cs [hunk 6]
```diff
@@ -153,6 +168,12 @@ public bool MatchDirectoryOnDisk(string dirPath, bool compareFileContents = true
             return true;
         }
 
+        private static string RemoveWhitespace(string actual)
+        {
+            actual = new string(actual.ToCharArray().Where(ch => !char.IsWhiteSpace(ch)).ToArray());
```
Comment (troydai) [inline test/Microsoft.Dnx.CommonTestUtils/DirTree.cs]: How effective it is. Will the performance sucks when the file is really really huge?

**Pull request closed without merging**